module-system/module-implementations/sov-ibc/src/context.rs

```diff
@@ -221,105 +370,210 @@ where
     }
 
     fn store_packet_acknowledgement(
         &mut self,
         ack_path: &AckPath,
         ack_commitment: AcknowledgementCommitment,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc
+            .packet_ack_map
+            .set(ack_path, &ack_commitment, *self.working_set.borrow_mut());
+        Ok(())
     }
 
     fn delete_packet_acknowledgement(&mut self, ack_path: &AckPath) -> Result<(), ContextError> {
-        todo!()
+        self.ibc
+            .packet_ack_map
+            .delete(ack_path, *self.working_set.borrow_mut());
+        Ok(())
     }
 
     fn store_channel(
         &mut self,
         channel_end_path: &ChannelEndPath,
         channel_end: ChannelEnd,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc.channel_end_map.set(
+            channel_end_path,
+            &channel_end,
+            *self.working_set.borrow_mut(),
+        );
+        Ok(())
     }
 
     fn store_next_sequence_send(
         &mut self,
         seq_send_path: &SeqSendPath,
         seq: Sequence,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc
+            .send_sequence_map
+            .set(seq_send_path, &seq, *self.working_set.borrow_mut());
+        Ok(())
     }
 
     fn store_next_sequence_recv(
         &mut self,
         seq_recv_path: &SeqRecvPath,
         seq: Sequence,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc
+            .recv_sequence_map
+            .set(seq_recv_path, &seq, *self.working_set.borrow_mut());
+        Ok(())
     }
 
     fn store_next_sequence_ack(
         &mut self,
         seq_ack_path: &SeqAckPath,
         seq: Sequence,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc
+            .ack_sequence_map
+            .set(seq_ack_path, &seq, *self.working_set.borrow_mut());
+        Ok(())
     }
 
-    fn increase_channel_counter(&mut self) {
-        todo!()
+    fn increase_channel_counter(&mut self) -> Result<(), ContextError> {
+        let next_channel_counter = self
+            .ibc
+            .channel_counter
+            .get(*self.working_set.borrow_mut())
+            .ok_or(ChannelError::Other {
+                description: "Channel counter not found".to_string(),
+            })?
+            .checked_add(1)
+            .ok_or(ChannelError::CounterOverflow)?;
+
+        self.ibc
+            .channel_counter
+            .set(&next_channel_counter, *self.working_set.borrow_mut());
+
+        Ok(())
     }
 
     fn emit_ibc_event(&mut self, event: IbcEvent) {
-        todo!()
+        // Note: as an interim solution, we transform IBC events into Tendermint
+        // events to simplify the conversion process, avoiding the need for
+        // converting individual IBC event types into a key-value pair of `&str`
+        let tm_event = tendermint::abci::Event::try_from(event)
+            .expect("failed to convert IBC event to Tendermint event");
```

Comment:
As discussed, let's return a `Result` so that we can remove an `expect`

Reply:
[37b97e8](https://github.com/informalsystems/sovereign-sdk/pull/1/commits/37b97e8d94ad3e1a06d28742058a1efdcca667a2)